### PR TITLE
feat: add classifiers for python3.9 and remove for python3.5

### PR DESCRIPTION
**Pull request by HemangChothani**
Towards #292 

- nox.py file is generated by synthtool, so didn't include changes here.

### CODE REVIEW

**Comment by andrewsg** [comment]
Will this refuse to install on Python 2.7 or will it just no longer be recommended?

**Comment by tseaver** [comment]
@andrewsg The following will refuse to install on 2.7, at least with any recent setuptools:
```python
    python_requires=">=3.6",
```

This *definitely* needs to get merged as a breaking change.

**Comment by crwilcox** [comment]
I'd recommend waiting to remove Python 2 until other breaking changes. At this time supporting Python 2 isn't providing any drag, but I think once it is we can remove it. @andrewsg @frankyn do we know of any changes that are also breaking to bundle?

File: noxfile.py [hunk 1]
```diff
@@ -27,8 +27,8 @@
 BLACK_PATHS = ["docs", "google", "tests", "noxfile.py", "setup.py"]
 
 DEFAULT_PYTHON_VERSION = "3.8"
-SYSTEM_TEST_PYTHON_VERSIONS = ["2.7", "3.8"]
-UNIT_TEST_PYTHON_VERSIONS = ["2.7", "3.5", "3.6", "3.7", "3.8"]
+SYSTEM_TEST_PYTHON_VERSIONS = ["3.8"]
+UNIT_TEST_PYTHON_VERSIONS = ["3.6", "3.7", "3.8", "3.9"]
```
Comment (tseaver) [inline noxfile.py:31]: Your comment says that you weren't including these changes, which should be driven by synthtool.

File: setup.py [hunk 2]
```diff
@@ -75,13 +75,11 @@
         "Intended Audience :: Developers",
         "License :: OSI Approved :: Apache Software License",
         "Programming Language :: Python",
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.7",
```
Comment (tseaver) [inline setup.py:79]: Dropping support for Python 2.7 is definitely a breaking change.  ISTM that #313 (where `synthtool` adds 3.9 support to `noxfile.py`, without dropping 3.7) suggests that we aren't yet ready to do that.  @crwilcox WDYT?

File: setup.py [hunk 3]
```diff
@@ -90,7 +88,7 @@
     namespace_packages=namespaces,
     install_requires=dependencies,
     extras_require=extras,
-    python_requires=">=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*",
+    python_requires=">=3.6",
```
Comment (tseaver) [inline setup.py:91]: Given we decide dropping 2.7 is premature:
```suggestion
    python_requires=">=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,!=3.5.*",
```

File: noxfile.py (hunk 1 above)
Comment (tseaver) [reply noxfile.py:31]: #313 indeed does update `noxfile.py` via synth.

Comment (tseaver) [reply noxfile.py]: Fixed when resolving conflict with #313.

File: setup.py (hunk 2 above)
Comment (tseaver) [reply setup.py:79]: I see that @crwilcox actually [commented already](https://github.com/googleapis/python-storage/pull/295#issuecomment-725695994) that we shouldn't drop 2.7 support.

**Review by tseaver — APPROVED** [review]

**Pull request merged**